src/components/settings/Settings.cpp

```diff
@@ -23,25 +32,228 @@ void Settings::SaveSettings() {
+
+  settings = bufferSettings;
+  CleanUpQcbor(&decodeCtx);
+  delete[] cborBuf;
 }
 
 void Settings::SaveSettingsToFile() {
   lfs_file_t settingsFile;
 
-  if ( fs.FileOpen(&settingsFile, "/settings.dat", LFS_O_WRONLY | LFS_O_CREAT) != LFS_ERR_OK) {
+  if (fs.FileOpen(&settingsFile, "/settings.cbor", LFS_O_WRONLY | LFS_O_CREAT) != LFS_ERR_OK) {
     return;
   }
-  fs.FileWrite(&settingsFile, reinterpret_cast<uint8_t*>(&settings), sizeof(settings));
+
+  UsefulBufC encodedSize;
+  encodedSize = Encode({NULL, SIZE_MAX});
+
+  UsefulBuf encodedCborBuf;
+  encodedCborBuf.len = encodedSize.len;
+  encodedCborBuf.ptr = new uint8_t[encodedSize.len]();
+
+  UsefulBufC encodedCbor = Encode(encodedCborBuf);
+
+  fs.FileWrite(&settingsFile, (uint8_t*)(encodedCbor.ptr), encodedCbor.len);
   fs.FileClose(&settingsFile);
+
+  delete[] (uint8_t*)(encodedCbor.ptr);
 }
+
+UsefulBufC Settings::Encode(UsefulBuf buffer) {
+  QCBOREncodeContext encodeCtx;
+  QCBOREncode_Init(&encodeCtx, buffer);
+
+  QCBOREncode_OpenMap(&encodeCtx);
+
+  QCBOREncode_AddUInt64ToMap(&encodeCtx, "StepsGoal", settings.stepsGoal);
+  QCBOREncode_AddUInt64ToMap(&encodeCtx, "ScreenTimeout", settings.screenTimeOut);
+  
+  QCBOREncode_AddUInt64ToMap(&encodeCtx, "ClockType", static_cast<uint64_t>(settings.clockType));
+  QCBOREncode_AddUInt64ToMap(&encodeCtx, "NotificationStatus", static_cast<uint64_t>(settings.notificationStatus));
+
+  QCBOREncode_AddUInt64ToMap(&encodeCtx, "ClockFace", settings.clockFace);
+  QCBOREncode_AddUInt64ToMap(&encodeCtx, "ChimesOption", static_cast<uint64_t>(settings.chimesOption));
+
+  QCBOREncode_OpenMapInMap(&encodeCtx, "PTS");
+  QCBOREncode_AddUInt64ToMap(&encodeCtx, "ColorTime", static_cast<uint64_t>(settings.PTS.ColorTime));
+  QCBOREncode_AddUInt64ToMap(&encodeCtx, "ColorBar", static_cast<uint64_t>(settings.PTS.ColorBar));
+  QCBOREncode_AddUInt64ToMap(&encodeCtx, "ColorBG", static_cast<uint64_t>(settings.PTS.ColorBG));
+  QCBOREncode_CloseMap(&encodeCtx);
+
+  QCBOREncode_AddUInt64ToMap(&encodeCtx, "WakeupMode", settings.wakeUpMode.to_ulong());
+  QCBOREncode_AddUInt64ToMap(&encodeCtx, "ShakeWakeThreshold", settings.shakeWakeThreshold);
+  QCBOREncode_AddUInt64ToMap(&encodeCtx, "BrightLevel", static_cast<uint64_t>(settings.brightLevel));
+
+  QCBOREncode_CloseMap(&encodeCtx);
+
+  UsefulBufC encodedCbor;
+  QCBOREncode_Finish(&encodeCtx, &encodedCbor);
+
+  return encodedCbor;
+}
+
+void Settings::MigrateSettingsToCBOR() {
+  lfs_info oldSettingsInfo;
+  if (fs.Stat("/settings.dat", &oldSettingsInfo) != LFS_ERR_OK) {
+    return;
+  }
+
+  SettingsData bufferSettings;
+  lfs_file_t settingsFile;
+
+  if (fs.FileOpen(&settingsFile, "/settings.dat", LFS_O_RDONLY) != LFS_ERR_OK) {
+    return;
+  }
+  fs.FileRead(&settingsFile, reinterpret_cast<uint8_t*>(&bufferSettings), sizeof(settings));
+  fs.FileClose(&settingsFile);
+
+  settings = bufferSettings;
+
+  SaveSettingsToFile();
+
+  fs.FileDelete("/settings.dat");
+}
```

Comment:
please add a newline at the end of source files
```suggestion
}

```